core/domain/state_domain.py

```diff
@@ -977,6 +2078,29 @@ def validate(
             raise utils.ValidationError(
                 'Hint(s) must be specified if solution is specified')
 
+        # These validations should only be run when we are validating for
+        # an exploration because as of now(16 Sept 2022) the data related
+        # to Question model is not yet fixed.
+        if validation_from_exploration:
+            if self.id == 'Continue':
+                self._validate_continue_input()
+            elif self.id == 'EndExploration':
+                self._validate_end_exploration_input()
+            elif self.id == 'NumericInput':
+                self._validate_numeric_input()
+            elif self.id == 'FractionInput':
+                self._validate_fraction_input()
+            elif self.id == 'NumberWithUnits':
+                self._validate_number_with_units_input()
+            elif self.id == 'MultipleChoiceInput':
+                self._validate_multi_choice_input()
+            elif self.id == 'ItemSelectionInput':
+                self._validate_item_selec_input()
+            elif self.id == 'DragAndDropSortInput':
+                self._validate_drag_and_drop_input()
+            elif self.id == 'TextInput':
+                self._validate_text_input()
```

Comment:
Can we use dictionary for this?

Reply:
Done

Reply:
I tried creating dictionary but it statrted to misbehave, as if I'm adding `Continue` interaction the validation for `End` interaction is running. So I changed it back.
Thanks

Reply:
Why it started misbehaving?

Reply:
Honestly, I'm not sure. I will get back to you on this.

Reply:
core/domain/state_domain.py